dynamo/plot/scatters.py

```diff
@@ -429,39 +496,55 @@ def _plot_basis_layer(cur_b, cur_l):
         #         )
         # else:
         #     continue
+
         for cur_c in color:
-            cur_title = cur_c
-            if cur_l in ["protein", "X_protein"]:
-                _color = adata.obsm[cur_l].loc[cur_c, :]
+            if not stack_colors:
+                cur_title = cur_c
             else:
-                _color = adata.obs_vector(cur_c, layer=None) if cur_l == "X" else adata.obs_vector(cur_c, layer=cur_l)
+                cur_title = stack_colors_title
+            _color = _get_adata_color(adata, cur_l, cur_c)
+
+            # select data rows based on stack color thresholding
+            if stack_colors:
+                _adata = adata[_color > stack_colors_threshold]
+                if values:
+                    _values = values[_color > stack_colors_threshold]
+                _color = _color[_color > stack_colors_threshold]
+                main_debug("stack colors: _adata len after thresholding by color value: %d" % (len(_adata)))
+                if len(_color) == 0:
+                    main_info("skipping color %s because no point of %s is above threshold" % (cur_c, cur_c))
+                    continue
+            else:
+                _adata = adata
```

Comment:
as I said above, make sure plotting all cells but set cells with value less than the threshold to lightgray

Reply:
I will modify the 515 main info.
This secondary branch is for non stack_colors case.